infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/SessionManager.java

```diff
@@ -581,5 +626,23 @@ public boolean isConnectionReceiver() {
     public void notifyLeadershipChange() {
         incomingMsgHandler.leadershipChanged();
     }
+
+
+    public void sendOutSessionToSinkSide(LogReplication.ReplicationSubscriber subscriber) {
+        Set<LogReplicationSession> sessionForSinkSide = this.createOutgoingSessionsBySubscriber(subscriber);
+        for (LogReplicationSession session : sessionForSinkSide) {
+            LogReplication.LogReplicationSinkSessionInitializationMsg msg =
+                    LogReplication.LogReplicationSinkSessionInitializationMsg.newBuilder().setSession(session).build();
+            CorfuMessage.RequestPayloadMsg payload =
+                    CorfuMessage.RequestPayloadMsg.newBuilder().setLrSinkSessionInitialization(msg).build();
+
+            updateRouterWithNewSessions();
+            createSourceFSMs();
+            logNewlyAddedSessionInfo();
+            topology.getRemoteClusterEndpoints().put(session.getSourceClusterId(), topology.getRemoteSourceClusters().get(session.getSourceClusterId()));
+            router.connect(router.getSessionToRemoteClusterDescriptor().get(session), session);
```

Comment:
cont of above comment:: We should not be blindly connecting to the remote cluster. 
Instead, we should check topology.getRemoteClusterEndpoints, and if the sink cluster is present in it, only then connect to remote. Otherwise we wait for the remote to connect to us.

like in this method: connectToRemoteClusters().